src/com/tictacgo/DirectionPickerFragment.java

```diff
@@ -0,0 +1,141 @@
+package com.tictacgo;
+
+import android.app.Fragment;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout.LayoutParams;
+import android.widget.ImageView;
+import android.widget.TableRow;
+
+import com.tictacgo.data.Board.Player;
+
+public class DirectionPickerFragment extends Fragment {
+  public static final String playerArgumentKey = "player";
+  public static final String gravityArgumentKey = "gravity";
+  public static final String heightArgumentKey = "height";
+
+  private Player player;
+  private int gravity;
+  private int height;
+
+  private View.OnClickListener directionClicked;
+
+  private OnDirectionPickedListener listener;
+
+  // Container Activity must implement this interface
+  public interface OnDirectionPickedListener {
+    public void onDirectionPicked(int dirx, int diry, int gravity);
+  }
+
+  /**
+   * Creates and returns a new DirectionPicker object.
+   *
+   * @param player The player whose turn it is.
+   * @param gravity The gravity of the location that was just pressed.
+   * @param height The height of the board.
+   * @return A new DirectionPicker object with the given parameters.
+   */
+  public static DirectionPickerFragment getInstance(Player player, int gravity, int height) {
+    Bundle arguments = new Bundle();
+    arguments.putSerializable(playerArgumentKey, player);
+    arguments.putInt(gravityArgumentKey, gravity);
+    arguments.putInt(heightArgumentKey, height);
+
+    DirectionPickerFragment directionPicker = new DirectionPickerFragment();
+    directionPicker.setArguments(arguments);
+    return directionPicker;
+  }
+
```

Comment:
This needs a public empty constructor, I think. The system uses it when it stops and restarts the app. We would then load fields with saveInstanceState and loadInstanceState. I'm waiting to do those methods when we implement phone rotation because that's when it's easy to test, and that's when it will mainly come into play. See the first answer here, for example: http://stackoverflow.com/questions/9245408/best-practice-for-instantiating-a-new-android-fragment